api/grpcserver/v2alpha1/transaction.go

```diff
@@ -377,6 +381,18 @@ func convertTxState(tx *types.MeshTransaction) *spacemeshv2alpha1.TransactionSta
 		state := spacemeshv2alpha1.TransactionState_TRANSACTION_STATE_PROCESSED
 		return &state
 	default:
+		evicted, err := s.conState.HasEvicted(tx.ID)
+		if err != nil {
+			ctxzap.Debug(ctx, "failed to check if tx is evicted",
+				zap.String("tx_id", tx.ID.String()),
+				zap.Error(err),
+			)
+			return nil
```

Comment:
I'm not sure if returning `nil` is correct here. The transaction then has no state - instead I think it should be something like unknown/unspecified/error - maybe `TransactionState_TRANSACTION_STATE_UNSPECIFIED`?